functional/declarative.md

```diff
@@ -0,0 +1,64 @@
+# Declarative vs Imperative
```

Comment:
Would call it `fp-paradigms.md` and the headline `Function Programming paradigms` as well. In the `SUMMARY` I would call it just `Paradigms` as it's already a subsection of FP.

Reply:
thoughts? @MarcoIeni

Reply:
if you do it like that it looks like "imperative" is a fp paradigm

Reply:
Anyway, I agree that current names are not that good

Reply:
Ah read it yesterday and forgot to answer, sorry. What would be your proposal?

Reply:
The headline and summary title could be `Declarive paradigm`?

Reply:
No, this should be called paradigms. If we put `Declarative paradigm` here and we talk about imperative, it is a bit confusing, so better just put the general world that covers both.

Reply:
Ok, should we go with `Programming paradigms`? We can also link [this](https://en.wikipedia.org/wiki/Programming_paradigm) in the introduction

Reply:
simonsan, tell me if it looks ok now :)

Reply:
reviewed it